### ISSUE TITLE
Check rust lints when an unknown lint is detected

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust-clippy#11855]
### Description

Suppose we have a lint section that looks like this:

```toml
[lints.clippy]
missing_docs = "warn"
```
Then, clippy will emit:

```
warning[E0602]: unknown lint: `clippy::missing_docs`
  |
  = help: did you mean: `clippy::erasing_op`
  = note: requested on the command line with `-W clippy::missing_docs`
  = note: `#[warn(unknown_lints)]` on by default
```

Instead, clippy could check the lints provided by Rust, and provide a more helpful suggestion: that the lint is actually a Rust lint.

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by y21** [comment]
I'm not sure if this is something that can be fixed from clippy's side. `unknown_lints` is a rustc lint. Clippy doesn't emit this warning.

**Comment by alice-i-cecile** [comment]
Sounds good, I'll open a duplicate on the Rust repo instead :)
